ra/ra.go

```diff
@@ -528,6 +528,7 @@ func (ra *RegistrationAuthorityImpl) NewRegistration(ctx context.Context, reques
 	}
 
 	// Check that contacts conform to our expectations.
+	// TODO(#8199): Remove this when no contacts are included in any requests.
```

Comment:
I'd like to see this limit included in this PR. Even with a request size limit of 50KB, you could still cram over 2000 contacts that look like `mailto:a@a.aa` into one new-account request.

Reply:
Done. I've also added (well, mostly copied from the RA, and modernized) unit tests for this behavior, and the other checks.